src/test/java/com/rybka/repository/CurrencyHistoryRepositoryTest.java

```diff
@@ -57,17 +55,41 @@ public void testOnCorrectHistoryFindTop5() {
         var historyOtherList = List.of(
                 buildRandomHistory(), buildRandomHistory(), buildRandomHistory()
         );
-        repository.saveAll(historyOtherList);
-        repository.saveAll(historyTop5List);
+
+        historyOtherList
+                .forEach(history -> {
+                    try {
+                        Thread.sleep(1000);
+                        repository.save(history);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                });
+
+        historyTop5List
+                .forEach(history -> {
+                    try {
+                        Thread.sleep(1000);
+                        repository.save(history);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                });
+
+        var expectedTop5HistoryList = repository.findAll();
 
         // when
-        var actualResultList = repository.findTop5ByOrderByIdDesc();
+        var actualResultList = repository.findTop5ByOrderByDateDesc();
 
         // then
-        var expectedHistoryList = historyTop5List.stream()
-                .sorted(Comparator.comparingInt(CurrencyHistory::getId).reversed())
+        var expectedHistoryList = expectedTop5HistoryList.stream()
```

Comment:
No need to build own list it can be created by yourself in the //given block. 
But sometimes we have to do like you have done.

Reply:
fixed